src/mango_native_proc.erl

```diff
@@ -283,11 +283,20 @@ get_text_field_type(_) ->
 
 make_text_field(TAcc, Type, Value) ->
     FieldName = make_text_field_name(TAcc#tacc.path, Type),
-    Fields = TAcc#tacc.fields,
-    case Fields == all_fields orelse lists:member(FieldName, Fields) of
+    Fields = case TAcc#tacc.fields of
```

Comment:
Move this case up to around line 152 as a function so that you only do the escape/join once per document. Ideally we should move this to the handle_call for <<"add_fun">> but it looks a bit difficult to make the detection there.